torchtune/utils/multi_dataset.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Tuple
+
+from omegaconf import DictConfig
+from torch.utils.data import Dataset
+from torchtune import config, utils
+from tqdm import tqdm
+
+log = utils.get_logger("DEBUG")
+
+
+class MultiDataset(Dataset):
+    def __init__(self, datasets: DictConfig[DictConfig], tokenizer):
+        self._datasets = datasets
+        self._tokenizer = tokenizer
+        self._data = []
+
+        # Load all datasets one by one
+        for dataset in datasets:
+            self._data.extend(self._loading_dataset(dataset))
```

Comment:
This means that all the data will iterate sequentially, right? What if I wanted to interleave the datasets and jump around during training instead of exhausting one dataset at time? Would the `DistributedSampler` handle this?

Reply:
Yeah, you right, i've changed this logic a new one, please refer MultiDataset class.